internal/build/tar_test.go

```diff
@@ -17,18 +17,22 @@ import (
 
 func TestArchiveDf(t *testing.T) {
 	f := newFixture(t)
-	ab := NewArchiveBuilder(model.EmptyMatcher)
-	defer ab.close()
-	defer f.tearDown()
 
 	dfText := "FROM alpine"
-	df := dockerfile.Dockerfile(dfText)
-	err := ab.archiveDf(f.ctx, df)
-	if err != nil {
-		t.Fatal(err)
-	}
+	pr, pw := io.Pipe()
+	go func() {
```

Comment:
for tests I think we could just use a `bytes.Buffer` for the `Writer` and skip the pipe and goroutine